cmd/addon/update.go

```diff
@@ -19,51 +21,63 @@ var (
 
 func init() {
 	updateCmd.Flags().StringVar(&addonFlags.Description, "description", addonFlags.Description, "The description of add-on")
-	updateCmd.Flags().StringVar(&addonFlags.FromFile, "from", addonFlags.FromFile, "The manifest file path of add-on")
-	updateCmd.Flags().StringVar(&addonFlags.Manifest, "manifest", addonFlags.Manifest, "The manifest file content of add-on, need to be base64 encode")
+	updateCmd.Flags().StringVarP(&addonFlags.FromFile, "from", "f", addonFlags.FromFile, "The manifest file path of add-on")
 	updateCmd.Flags().StringToStringVar(&addonFlags.Values, "set", addonFlags.Values, "Set value to replace parameters defined in manifest")
+	updateCmd.Flags().StringArrayVar(&addonFlags.RemoveValues, "unset", addonFlags.RemoveValues, "The values of the add-on to unset, will ignore if the value is not present in the list")
 }
 
 func UpdateCmd() *cobra.Command {
+	updateCmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		if addonFlags.FromFile != "" && !utils.IsFileExists(addonFlags.FromFile) {
+			return fmt.Errorf("manifest file %s is not exist", addonFlags.FromFile)
+		}
+		return nil
+	}
 	updateCmd.Run = func(cmd *cobra.Command, args []string) {
 		name := args[0]
 		addon, err := common.DefaultDB.GetAddon(name)
 		if err != nil {
 			logrus.Fatalln(err)
 		}
-		var manifestFile string
-		manifestContent := string(addon.Manifest)
+		var manifestString string
+		isChanged := false
 		if addonFlags.FromFile != "" {
-			manifestFile = addonFlags.FromFile
-		} else if addonFlags.Manifest != "" {
-			manifestContent = addonFlags.Manifest
+			manifestString, err = common.GetManifest(addonFlags.FromFile, "")
+			if err != nil {
+				logrus.Error(err)
+				return
+			}
 		}
-
-		manifestString, err := common.GetManifest(manifestFile, manifestContent)
-		if err != nil {
-			logrus.Fatalln(err)
+		if manifestString != "" && manifestString != string(addon.Manifest) {
+			addon.Manifest = []byte(manifestString)
+			isChanged = true
 		}
-		manifest := []byte(manifestString)
 
-		isChanged := false
 		if addonFlags.Description != "" && addonFlags.Description != addon.Description {
 			addon.Description = addonFlags.Description
 			isChanged = true
 		}
 
-		if !bytes.Equal(manifest, addon.Manifest) {
-			addon.Manifest = manifest
-			isChanged = true
+		if addon.Values == nil {
+			addon.Values = make(types.StringMap)
+		}
+		// unset values
+		for _, v := range addonFlags.RemoveValues {
+			if _, ok := addon.Values[v]; ok {
+				delete(addon.Values, v)
+			}
+		}
+		// re-set values
+		for k, v := range addonFlags.Values {
+			addon.Values[k] = v
 		}
-
 		if !reflect.DeepEqual(addonFlags.Values, addon.Values) {
```

Comment:
I think it needs to copy for value of `addon.Values`. After unset values and re-set values, deep equal to check the copy one and the origin one instead of `addonFlags.Values`. If users update add-on without setting `--set` args. The `isChanged` will always be true here even though there's nothing to change.

Reply:
Updated